keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/BakeryLifecycleMonitor.kt

```diff
@@ -57,25 +60,15 @@ class BakeryLifecycleMonitor(
           else -> publishUnknownEvent(task, execution)
         }
 
-        if (task.numFailures > 0) {
-          // we only care about consecutive failures, and we just had a success
-          monitorRepository.clearFailuresGettingStatus(task)
-        }
-
         if (execution.status.isComplete()) {
           endMonitoringOf(task)
+        } else {
+          markSuccessFetching(task)
```

Comment:
I personally dislike method names that rely on good parameter names to understand what they do. I'd suggest `markSuccessFetchingTask` instead.

Reply:
We will compromise on `markSuccessFetchingStatus` and `handleFailureFetchingStatus`